include/mbgl/storage/response.hpp

```diff
@@ -35,7 +35,7 @@ class Response {
     optional<std::string> etag;
 
     bool isFresh() const {
-        return !expires || *expires > util::now();
+        return (!expires || *expires > util::now()) && !error;
```

Comment:
> I think they are logically the same. But yours is definitely easier to read.

In the case that `error` and `expires` are both set, `(!expires || *expires > util::now()) && !error` is always false, but `expires ? *expires > util::now() : !error` compares `expires` to the current time and ignores `error`. I'm arguing that the latter is the desired behavior.

> Currently NotFound doesn't trigger a retry which makes sense, specially for tiles.

I don't think this is the case. Imagine the scenario where someone uploads a tileset for a small area, and then expands it to cover a larger area. If you cached a 404 for tiles outside the original boundary, you'd want to be able to obtain the newly available tiles when that cached 404 expires.